haskell/test/Test/Ligo/BaseDAO/Integrational/Proposal.hs

```diff
@@ -93,3 +104,196 @@ checkFreezeHistoryTracking  = do
                   }
             when (fh /= (Just expected)) $
               Left $ CustomTestError "BaseDAO contract did not unstake tokens after voting period"
+
+checkQuorumThresholdDynamicUpdate :: IntegrationalScenarioM ()
+checkQuorumThresholdDynamicUpdate  = do
+  now <- use isNow
+
+  withOriginated 2 (\(admin:_) -> do
+    tokenContract <- lOriginate dummyFA2Contract "TokenContract" [] (toMutez 0)
+    pure $ mkFullStorage
+      ! #admin admin
+      ! #votingPeriod 10
+      ! #quorumThreshold (mkQuorumThreshold 3 10)
+      ! #extra dynRecUnsafe
+      ! #metadata mempty
+      ! #now now
+      ! #tokenAddress (unTAddress tokenContract)
+      ! #maxChangePercent 19
+      ! #changePercent 5
+      ! #governanceTotalSupply 100
+      ! #customEps mempty
+    ) $ \(_:wallet1:_) baseDao -> do
+        let requiredFrozen = (10 :: Natural)
+
+        let proposalMeta1 = ""
+
+        tTransfer  (#from .! wallet1) (#to .! (unTAddress baseDao)) zeroMutez (unsafeBuildEpName "freeze") (toVal (2 * requiredFrozen))
+
+        rewindTime 11
+        -- First proposal period, cycle 0
+
+        tTransfer  (#from .! wallet1) (#to .! (unTAddress baseDao)) zeroMutez (unsafeBuildEpName "propose") (toVal (ProposeParams requiredFrozen proposalMeta1))
+
+        lExpectStorage @FullStorage baseDao $ \storage -> do
+            let qs = sQuorumThresholdAtCycle $ fsStorage storage
+            let expected = QuorumThresholdAtCycle 1 (mkQuorumThreshold 3 10) 10 -- We don't expect the inital quorum to have updated here
+            when (qs /= expected) $
+              Left $ CustomTestError ("BaseDAO contract updated quorum threshold unexpectedly" <> (show qs))
+
+        -- skip this proposal period and next voting period to be in next proposal period
+        rewindTime 21 -- cycle 2
+
+        let proposalMeta2 = "A"
+        tTransfer  (#from .! wallet1) (#to .! (unTAddress baseDao)) zeroMutez (unsafeBuildEpName "propose") (toVal (ProposeParams requiredFrozen proposalMeta2))
+
+        lExpectStorage @FullStorage baseDao $ \storage -> do
+            let qs = sQuorumThresholdAtCycle $ fsStorage storage
+            -- We start with quorumThreshold of 3/10, with participation as 10 and governance total supply as 100
+            -- participation = 10/100 = 1/10
+            -- so possible_new_quorum = 3/10 * (1 - 0.05)  + (1/10 * 5/100)
+            -- so 3/10 * (19/20) + (5/1000) = 57/200 + 1/200 = 58/200 = 29/100
+            let expected = QuorumThresholdAtCycle 2 (mkQuorumThreshold 290 1000) 10
```

Comment:
here (and in following places) these tests rely on these calculations to be correct. The comments are good, however this makes updating the tests on small changes quite complex, wouldn't it be possible to make a function to calculate the expected new threshold from some given arguments?

Reply:
Have updated.